Makefile

```diff
@@ -107,160 +69,196 @@ ifneq ($(DASHBOARD_DISTRIBUTION_DIR),)
+
+DEADLOCK_ENABLE := $$(\
+						find . -name "*.go" \
+						| xargs -n 1 sed -i.bak 's/sync\.RWMutex/deadlock.RWMutex/;s/sync\.Mutex/deadlock.Mutex/' && \
+						find . -name "*.go" | xargs grep -lE "(deadlock\.RWMutex|deadlock\.Mutex)" \
+						| xargs goimports -w)
+DEADLOCK_DISABLE := $$(\
+						find . -name "*.go" \
+						| xargs -n 1 sed -i.bak 's/deadlock\.RWMutex/sync.RWMutex/;s/deadlock\.Mutex/sync.Mutex/' && \
+						find . -name "*.go" | xargs grep -lE "(sync\.RWMutex|sync\.Mutex)" \
+						| xargs goimports -w && \
+						find . -name "*.bak" | xargs rm && \
+						go mod tidy)
+
+deadlock-enable: install-tools
+	# Enabling deadlock...
+	@$(DEADLOCK_ENABLE)
+
+deadlock-disable: install-tools
+	# Disabling deadlock...
+	@$(DEADLOCK_DISABLE)
+
+failpoint-enable: install-tools
+	# Converting failpoints...
+	@$(FAILPOINT_ENABLE)
+
+failpoint-disable: install-tools
+	# Restoring failpoints...
+	@$(FAILPOINT_DISABLE)
+
+.PHONY: deadlock-enable deadlock-disable failpoint-enable failpoint-disable
+
+#### Test ####
+
+PACKAGE_DIRECTORIES := $(subst $(PD_PKG)/,,$(PACKAGES))
+TEST_PKGS := $(filter $(shell find . -iname "*_test.go" -exec dirname {} \; | \
+                     sort -u | sed -e "s/^\./github.com\/tikv\/pd/"),$(PACKAGES))
+BASIC_TEST_PKGS := $(filter-out $(PD_PKG)/tests%,$(TEST_PKGS))
+
+test: install-tools
 	# testing all pkgs...
 	@$(DEADLOCK_ENABLE)
 	@$(FAILPOINT_ENABLE)
-	CGO_ENABLED=1 GO111MODULE=on go test -tags tso_function_test -timeout 20m -race -cover $(TEST_PKGS) || { $(FAILPOINT_DISABLE); $(DEADLOCK_DISABLE); exit 1; }
+	CGO_ENABLED=1 go test -tags tso_function_test -timeout 20m -race -cover $(TEST_PKGS) || { $(FAILPOINT_DISABLE); $(DEADLOCK_DISABLE); exit 1; }
 	@$(FAILPOINT_DISABLE)
 	@$(DEADLOCK_DISABLE)
 
-basic-test:
+basic-test: install-tools
 	# testing basic pkgs...
 	@$(FAILPOINT_ENABLE)
-	GO111MODULE=on go test $(BASIC_TEST_PKGS) || { $(FAILPOINT_DISABLE); exit 1; }
+	go test $(BASIC_TEST_PKGS) || { $(FAILPOINT_DISABLE); exit 1; }
 	@$(FAILPOINT_DISABLE)
 
-test-with-cover: install-go-tools dashboard-ui
+test-with-cover: install-tools dashboard-ui
 	# testing all pkgs (expect TSO consistency test) with converage...
 	@$(FAILPOINT_ENABLE)
 	for PKG in $(TEST_PKGS); do\
 		set -euo pipefail;\
-		CGO_ENABLED=1 GO111MODULE=on go test -race -covermode=atomic -coverprofile=coverage.tmp -coverpkg=./... $$PKG 2>&1 | grep -v "no packages being tested" && tail -n +2 coverage.tmp >> covprofile || { $(FAILPOINT_DISABLE); rm coverage.tmp && exit 1;}; \
+		CGO_ENABLED=1 go test -race -covermode=atomic -coverprofile=coverage.tmp -coverpkg=./... $$PKG 2>&1 | grep -v "no packages being tested" && tail -n +2 coverage.tmp >> covprofile || { $(FAILPOINT_DISABLE); rm coverage.tmp && exit 1;}; \
 		rm coverage.tmp;\
 	done
 	@$(FAILPOINT_DISABLE)
 
-# The command should be used in daily CI，it will split some tasks to run parallel.
-# It should retain report.xml,coverage,coverage.xml and package.list to analyze.
```

Comment:
why remove it，it will run in ci-pipeline.

Reply:
I didn't find anywhere using it. I think now coverage is reported by `test-with-cover`. You can see this PR is normally generating coverage report.

Reply:
add back.